pkg/db/query/test_queries.go

```diff
@@ -117,3 +157,26 @@ FROM results;
 	log.Infof("TestReportExcludeVariants completed in %s", elapsed)
 	return testReport, nil
 }
+
+// TestsByNURPAndStandardDeviation returns a test report for every test in the db matching the given substrings, separated by variant.
+func TestsByNURPAndStandardDeviation(dbc *db.DB, release, table string) *gorm.DB {
+	stats := dbc.DB.Table(table).
+		Select(`
+                 id                                                                             AS test_id,
+                 avg((current_successes + current_flakes) * 100.0 / NULLIF(current_runs, 0))    AS working_average,
+                 stddev((current_successes + current_flakes) * 100.0 / NULLIF(current_runs, 0)) AS working_standard_deviation`).
+		Where(`release = ?`, release).
+		Group("id")
+
+	passRates := dbc.DB.Table(table).
+		Select(`id as test_id, variants as pass_rate_variants, `+QueryTestPercentages).
+		Where(`release = ?`, release)
+
+	return dbc.DB.
+		Table(table).
+		Select("*, (current_working_percentage - working_average) as delta_from_average").
+		Joins(fmt.Sprintf(`INNER JOIN (?) as pass_rates on pass_rates.test_id = %s.id AND pass_rates.pass_rate_variants = %s.variants`, table, table), passRates).
+		Joins(fmt.Sprintf(`JOIN (?) as stats ON stats.test_id = %s.id`, table), stats).
+		Where(`release = ?`, release).
+		Where(fmt.Sprintf("NOT ('never-stable'=any(%s.variants))", table))
```

Comment:
Have you tested this will a full copy of the prod db from sippy-exports in s3 Ken? I'm curious about how fast it is. I tried locally with my last import and it seems totally fine.

And my followup question, would it be possible to also be able to do this such that we could also see the oddities in just passing percentage, and especially flaking percentage. (i.e. the three working views Stephen added for us on the tests table)

Reply:
@stbenjam does this sql look ok to you? It's a bit beyond my abilities.

Actually in that line of thinking, Ken could you expand a little on how the sql all fits together with a comment for people like me. :)

Reply:
It's ok. I think I wrote something like it in my original z-score PR (I like this version with deltas better).

+1 for showing passing and flaking.